src/store/addBook/actions.js

```diff
@@ -0,0 +1,95 @@
+import axios from "axios";
+import { apiUrl } from "../../config/constants";
+import selectUser from "../user/selectors";
+import {
+  appLoading,
+  appDoneLoading,
+  showMessageWithTimeout,
+  setMessage
+} from "../appState/actions";
+
+export const FETCH_AUTHOR_SUCCESS = "FETCH_AUTHOR_SUCCESS";
+export const FETCH_DESCRIPTION_SUCCESS = "FETCH_DESCRIPTION_SUCCESS";
+export const FETCH_IMAGE_SUCCESS = "FETCH_IMAGE_SUCCESS";
+export const POST_BOOK_SUCCESS = "POST_BOOK_SUCCESS ";
+
+export const addBook = (
+  title,
+  description,
+  imageUrl,
+  borrowingPeriod,
+  author,
+  language
+) => {
+  return async (dispatch, getState) => {
+    const state = getState();
+    console.log("STATE in ADDBOOK", state);
+    const token = state.user.token;
+    console.log(token, state.user);
+    try {
+      const response = await axios.post(
+        `${apiUrl}/books`,
+        {
+          title,
+          description,
+          imageUrl,
+          borrowingPeriod,
+          author,
+          language
+        },
+        {
+          headers: {
+            Authorization: `Bearer ${token}`
+          }
+        }
+      );
+      console.log(response.data);
+      dispatch(
+        showMessageWithTimeout(
+          "success",
+          false,
+          `${title} is added, thanks for sharing your book with others!`,
+          6000
+        )
+      );
+      dispatch({ type: "POST_BOOK_SUCCESSS", payload: response.data });
+      console.log("NEW BOOK", response.data);
+      dispatch(appDoneLoading());
+    } catch (e) {
+      // console.log("danger", e.response.data);
+      // dispatch(showMessageWithTimeout("danger", false, e.response.data, 3000));
+      dispatch(appDoneLoading());
+    }
+  };
+};
+
+export function fetchFromGoogle(title) {
```

Comment:
this action is already done in addbook component, please remove it from actions file.

Reply:
fixed